benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/tasks/abort/AbortRunningTestTask.java

```diff
@@ -0,0 +1,46 @@
+package cloud.benchflow.testmanager.tasks.abort;
+
+import cloud.benchflow.testmanager.BenchFlowTestManagerApplication;
+import cloud.benchflow.testmanager.exceptions.BenchFlowTestIDDoesNotExistException;
+import cloud.benchflow.testmanager.services.external.BenchFlowExperimentManagerService;
+import cloud.benchflow.testmanager.services.internal.dao.BenchFlowTestModelDAO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Jesper Findahl (jesper.findahl@gmail.com) created on 2017-07-02
+ */
+public class AbortRunningTestTask implements Runnable {
+
+  private static Logger logger =
+      LoggerFactory.getLogger(AbortRunningTestTask.class.getSimpleName());
+
+  private String testID;
+  private BenchFlowExperimentManagerService experimentManagerService;
+  private BenchFlowTestModelDAO testModelDAO;
+
+  public AbortRunningTestTask(String testID) {
+    this.testID = testID;
+    this.experimentManagerService = BenchFlowTestManagerApplication.getExperimentManagerService();
+    this.testModelDAO = BenchFlowTestManagerApplication.getTestModelDAO();
+  }
+
+  @Override
+  public void run() {
+
+    logger.info("running: " + testID);
+
+    try {
+
+      String experimentID = testModelDAO.getRunningExperiment(testID);
+
+      if (experimentID != null) {
+        experimentManagerService.abortBenchFlowExperiment(experimentID);
```

Comment:
You might want to get and `ACK` about the execution of the operation, because otherwise it is really complex to control the behaviour of the distributed system.